datalake/src/main/java/com/sequenceiq/datalake/service/sdx/stop/SdxStopService.java

```diff
@@ -100,4 +123,18 @@ public void stopAllDatahub(Long sdxId) {
         SdxCluster sdxCluster = sdxService.getById(sdxId);
         distroxService.stopAttachedDistrox(sdxCluster.getEnvCrn());
     }
+
+    public boolean checkIfStoppable(SdxCluster cluster) {
+        Optional<FlowLog> lastFlowLog = flowLogService.getLastFlowLog(cluster.getId());
+        if (lastFlowLog.isPresent()) {
+            String flowChainType = flowChainLogService.getFlowChainType(
+                    lastFlowLog.get().getFlowChainId()
```

Comment:
What if a flow is running which doesn't have flow chain id set because it is not a flow chain?

Reply:
Nevermind, I looked into the `getFlowChainType` implementation and it already does null checks, so it `flowChainType` should be `null` if the ID does not exist.

I will add a unit test for this though.